lib/board.rb

```diff
@@ -0,0 +1,73 @@
+class Board
+
+  attr_accessor :cells
+
+  def initialize
+    self.reset!
+  end
+
+  def reset!
+    self.cells = Array.new(9, " ")
+  end
+
+  def display
+    horiz_line = "-----------"
+    puts " #{self.cells[0]} | #{self.cells[1]} | #{self.cells[2]} "
+    puts horiz_line
+    puts " #{self.cells[3]} | #{self.cells[4]} | #{self.cells[5]} "
+    puts horiz_line
+    puts " #{self.cells[6]} | #{self.cells[7]} | #{self.cells[8]} "
+  end
+
+  # Accepts position as a player-perspective string (1-9)
+  def position(position)
+    index = position.to_i - 1
+    self.cells[index]
+  end
+
+  # Accepts position as a player-perspective string (1-9)
+  def update(position, player)
+    # put player.token in the correct position
+    self.cells[position.to_i - 1] = player.token
+  end
+
+  def full?
+    self.cells.all? do |element|
+      element != " " && element != "" && element != nil
+    end
+  end
+
+  def turn_count
+    counter = 0
+    self.cells.each do |element|
```

Comment:
Use the highest level and most expressive iterator possible to solve a problem. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  cells.count { |cell| cell == "X" || cell == "O" }
end
```